basic_autonomy/include/basic_autonomy/basic_autonomy.hpp

```diff
@@ -363,10 +363,34 @@ namespace basic_autonomy
          * \param speeds Velocity profile to shift. The first point should be the current vehicle speed
          * \param downtrack Distance points for each velocity point. Should have the same size as speeds and start from 0
          * \param response_lag The lag in seconds before which the vehicle will not meaningfully accelerate
-         * 
+         *
          * \return A Shifted trajectory
-         */ 
+         */
         std::vector<double> apply_response_lag(const std::vector<double>& speeds, const std::vector<double> downtracks, double response_lag);
+
+        /**
+         * \brief Applies a yield trajectory to the original trajectory set in response
+         * \param node_handler a node interface to use the logger and time
+         * \param req The service request containing the maneuvers to plan trajectories for and current vehicle state
+         * \param resp The original response containing the planned trajectory to be modified
+         * \param yield_client yield_client to call for
+         * \param yield_plugin_service_call_timeout yield client's timeout in milliseconds int
+         *
+         * \return The original response modified to contain the modified planned trajectory
+         */
+        carma_planning_msgs::srv::PlanTrajectory::Response::SharedPtr modify_trajectory_to_yield_to_obstacles(
+            std::shared_ptr<carma_ros2_utils::CarmaLifecycleNode> node_handler,
+            carma_planning_msgs::srv::PlanTrajectory::Request::SharedPtr req, carma_planning_msgs::srv::PlanTrajectory::Response::SharedPtr resp,
+            carma_ros2_utils::ClientPtr<carma_planning_msgs::srv::PlanTrajectory> yield_client, int yield_plugin_service_call_timeout);
```

Comment:
Looks like the `.cpp` file changes added the `const` qualifier but did not change the type to references.